docs/design/release_automation_design.md

```diff
@@ -0,0 +1,72 @@
+## Release Automation Process Design
```

Comment:
I think it'd be worth removing the section on automation until we're farther on the manual process; i.e., what steps we specifically want to have performed during a release.